pytest-embedded-idf/pytest_embedded_idf/unity_tester.py

```diff
@@ -450,37 +480,15 @@ def run_all_single_board_cases(
         if isinstance(name, str):
             name: t.List[str] = [name]
 
-        def validate_group(case_groups):
-            if not group:
-                return True
-
-            for _or in group:
-                for _and in _or:
-                    invert = _and.startswith('!')
-                    _and = _and.lstrip('!')
-                    result = _and in case_groups
-                    if invert:
-                        result = not result
-                    if not result:
-                        break
-                else:
-                    return True
-
-            return False
-
         for case in self.test_menu:
-            selected = False
-            if not group and not name and not attributes:
-                selected = True
-            if group and validate_group(case.groups):
-                selected = True
-            if name and case.name in name:
-                selected = True
-            if attributes and all(case.attributes.get(k) == v for k, v in attributes.items()):
-                selected = True
+            selected = self.__select_to_run(group, name, attributes, case.groups, case.name, case.attributes)
 
             if not selected:
                 continue
+            if dry_run:
+                logging.info('%s | %s', case.index, case.name)
```

Comment:
could you make it happen under the "ignore" block later?

this could be confusing when one test case is "[ignored]", showing here, but not been run later.